From ce6844d2df9420ff80ced37a23d88ce48d1f31d7 Mon Sep 17 00:00:00 2001
From: Simon Schricker <sschricker@suse.de>
Date: Thu, 14 Mar 2019 14:07:51 +0100
Subject: [PATCH] Change nvmefc-connect@.service to type `simple`

The idea is that udevd calls systemd for the nvme connect-all calls so
that it doesn't have to wait for it to return. But systemd waits for
services of type `oneshot` when called with systemctl start until the
process specified in the service file terminates.

Instead use type `simple`: systemd will return immediately after
starting the service, regardless of how long the nvme connect-all call
takes.

Signed-off-by: Simon Schricker <sschricker@suse.de>
---
 nvme-fc-autoconnect/nvmefc-connect@.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvme-fc-autoconnect/nvmefc-connect@.service b/nvme-fc-autoconnect/nvmefc-connect@.service
index 8465ec6..a8ff271 100644
--- a/nvme-fc-autoconnect/nvmefc-connect@.service
+++ b/nvme-fc-autoconnect/nvmefc-connect@.service
@@ -9,5 +9,5 @@ PartOf=nvmefc-connect.target
 Requires=nvmefc-connect.target
 
 [Service]
-Type=oneshot
+Type=simple
 ExecStart=/bin/sh -c "/usr/sbin/nvme connect-all --transport=fc `/usr/bin/echo -e '%i'`"
-- 
2.16.4

